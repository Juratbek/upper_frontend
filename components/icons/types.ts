export interface IIconProps {
  color?: string;
  variant?: 'outlined' | 'fulfilled';
  width?: number;
  height?: number;
  opacity?: number;
}
