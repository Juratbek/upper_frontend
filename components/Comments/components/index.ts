export * from './Comment/Comment';
export * from './Form/Form';
