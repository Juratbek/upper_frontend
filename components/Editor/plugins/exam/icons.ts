export const DeleteIcon = `
        <svg width="18" height="18" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
            <path d="M15.741 7.96179C15.741 14.6443 16.7029 17.665 10.233 17.665C3.76238 17.665 4.74413 14.6443 4.74413 7.96179" stroke="#D21D1D" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
            <path d="M16.971 5.39978H3.51221" stroke="#D21D1D" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
            <path d="M13.0957 5.39982C13.0957 5.39982 13.5362 2.26172 10.241 2.26172C6.94656 2.26172 7.38704 5.39982 7.38704 5.39982" stroke="#D21D1D" stroke-width="1.5" stroke-linecap="round" stroke-linejoin="round"/>
        </svg>
    `;
