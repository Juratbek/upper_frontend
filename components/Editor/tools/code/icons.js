export const CodeIcon = `
    <svg width="800px" height="800px" viewBox="0 0 24 24" fill="none" xmlns="http://www.w3.org/2000/svg">
        <path d="M9 21C7.89543 21 7 20.1046 7 19V15.3255C7 14.8363 7 14.5917 6.94474 14.3615C6.89575 14.1575 6.81494 13.9624 6.70528 13.7834C6.5816 13.5816 6.40863 13.4086 6.06274 13.0627L5 12L6.06274 10.9373C6.40864 10.5914 6.5816 10.4184 6.70528 10.2166C6.81494 10.0376 6.89575 9.84254 6.94474 9.63846C7 9.40829 7 9.1637 7 8.67452V5C7 3.89543 7.89543 3 9 3M15 21C16.1046 21 17 20.1046 17 19V15.3255C17 14.8363 17 14.5917 17.0553 14.3615C17.1043 14.1575 17.1851 13.9624 17.2947 13.7834C17.4184 13.5816 17.5914 13.4086 17.9373 13.0627L19 12L17.9373 10.9373C17.5914 10.5914 17.4184 10.4184 17.2947 10.2166C17.1851 10.0376 17.1043 9.84254 17.0553 9.63846C17 9.40829 17 9.1637 17 8.67452V5C17 3.89543 16.1046 3 15 3" stroke="#000000" stroke-width="2" stroke-linecap="round" stroke-linejoin="round"/>
    </svg>
`;

export const CopyIcon = `
    <svg width="20" height="20" viewBox="0 0 20 20" fill="none">
        <g id="Copy">
            <path id="Vector" d="M5.00037 5.10656L5 13.125C5 15.625 5.625 16.25 7.91953 16.2434L8.125 16.25L13.643 16.2509C13.3854 16.9787 12.6911 17.5 11.875 17.5H7.5C5 17.5 3.75 16.2434 3.75 13.75V6.875C3.75 6.05848 4.27193 5.36386 5.00037 5.10656ZM14.375 2.5C15.4105 2.5 16.25 3.33947 16.25 4.375V13.125C16.25 14.1605 15.4105 15 14.375 15H8.125C6.875 15 6.25 14.375 6.25 13.125V4.375C6.25 3.33947 7.08947 2.5 8.125 2.5H14.375ZM14.375 3.75H8.125C7.77982 3.75 7.5 4.02982 7.5 4.375V13.125C7.5 13.4702 7.77982 13.75 8.125 13.75H14.375C14.7202 13.75 15 13.4702 15 13.125V4.375C15 4.02982 14.7202 3.75 14.375 3.75Z" fill="white"/>
        </g>
    </svg>
`;

export const DoneIcon = `
    <svg width="1.4em" height="1.4em" viewBox="0 0 24 24" fill="none">
        <path d="M5 14L8.23309 16.4248C8.66178 16.7463 9.26772 16.6728 9.60705 16.2581L18 6" stroke="#33363F" stroke-width="2" stroke-linecap="round"/>
    </svg>
`;
