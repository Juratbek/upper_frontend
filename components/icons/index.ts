export * from './UpRight';
export * from './Done';
export * from './Down';
export * from './Copy';
export * from './Home';
export * from './Write';
export * from './User';
export * from './Settings';
export * from './Moon';
export * from './Notification';
export * from './Search';
export * from './Eye';
export * from './EyeSlash';
export * from './Telegram';
export * from './Facebook';
export * from './Like';
export * from './Dislike';
export * from './Comment';
export * from './Share';
