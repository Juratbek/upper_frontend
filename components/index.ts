export * from './Navigation/Navigation';
export * from './Sidebar/Sidebar';
export * from './TabsHeader/TabsHeader';
export * from './Article/Article';
export * from './Actions/Actions';
export * from './Label/Label';
export * from './Author/Author';
export * from './Blog/Blog';
export * from './Button/Button';
export * from './Divider/Divider';
export * from './SidebarArticle/SidebarArticle';
export * from './ArticleImg/ArticleImg';
export * from './SearchInput/SearchInput';
export * from './SidebarBlog/SidebarBlog';
export * from './Avatar/Avatar';
export * from './TabBody/TabBody';
export * from './CommentNotification/CommentNotification';
export * from './LikeNotification/LikeNotification';
export * from './Input/Input';
export * from './Select/Select';
