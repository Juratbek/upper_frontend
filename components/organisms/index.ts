export * from './auth-modal';
export * from './header';
export * from './navigation';
export * from './sidebar';
export * from './footer';
export * from './connect-telegram';
