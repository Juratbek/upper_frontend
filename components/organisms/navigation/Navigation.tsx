import { Divider } from 'components/lib';
import { useRouter } from 'next/router';
import { FC, Fragment } from 'react';
import { WEB_APP_ROOT_DIR } from 'variables';

import { NavItem } from './components/item';
import { INavigation } from './Navigation.types';

export const Navigation: FC<{ items?: INavigation[] }> = ({ items = [] }) => {
  const { pathname } = useRouter();

  return (
    <>
      {items.map((item, index) => {
        const isActive = WEB_APP_ROOT_DIR + item.href === pathname;
        return (
          <Fragment key={item.text}>
            <NavItem {...item} active={isActive} key={item.text} />
            {index + 1 !== items.length && <Divider color='tertiary' my={4} />}
          </Fragment>
        );
      })}
    </>
  );
};
