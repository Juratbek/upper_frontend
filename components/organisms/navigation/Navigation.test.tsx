import { render } from '@testing-library/react';
import { HomeIcon, NotificationIcon, WriteIcon } from 'components/icons';

import { Navigation } from './Navigation';
import { INavigation } from './Navigation.types';

const mock: INavigation[] = [
  {
    href: '/home',
    text: 'Home',
    icon: HomeIcon,
  },
  {
    href: '/notifications',
    text: 'Notifications',
    badge: '+9',
    icon: NotificationIcon,
  },
  {
    href: '/articles',
    text: 'Articles',
    icon: WriteIcon,
  },
];

describe('Navigation', () => {
  it('render', () => {
    render(<Navigation items={mock} />);
    expect(document.body).toMatchSnapshot();
  });
});
