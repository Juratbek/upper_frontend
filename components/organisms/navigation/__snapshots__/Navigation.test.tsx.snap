// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Navigation > render 1`] = `
<body>
  <div>
    <a
      class="root link"
      href="/web/home"
    >
      <svg
        fill="none"
        height="20"
        role="icon"
        viewBox="0 0 24 24"
        width="20"
      >
        <path
          d="M9 17V21.9997H5C3.89543 21.9997 3 21.1043 3 19.9997V10.0332C3 9.73801 3.13046 9.45786 3.35643 9.26786L10.7129 3.08228C11.4569 2.45664 12.5431 2.45665 13.2871 3.08228L20.6436 9.26786C20.8695 9.45786 21 9.73801 21 10.0332V20C21 21.1046 20.1046 22 19 22H15V17C15 13 9 13 9 17Z"
          fill="black"
        />
      </svg>
      <p
        class="m-0 flex-1"
      >
        Home
      </p>
    </a>
    <a
      class="root link"
      href="/web/notifications"
    >
      <svg
        fill="none"
        height="20"
        role="icon"
        viewBox="0 0 24 24"
        width="20"
      >
        <mask
          fill="white"
          id="path-2-inside-1_630_4298"
        >
          <path
            d="M4.5 10.5C4.5 6.35786 7.85786 3 12 3C16.1421 3 19.5 6.35786 19.5 10.5V17C19.5 17.5523 19.0523 18 18.5 18H5.5C4.94772 18 4.5 17.5523 4.5 17V10.5Z"
          />
        </mask>
        <path
          d="M4.5 10.5C4.5 6.35786 7.85786 3 12 3C16.1421 3 19.5 6.35786 19.5 10.5V17C19.5 17.5523 19.0523 18 18.5 18H5.5C4.94772 18 4.5 17.5523 4.5 17V10.5Z"
          mask="url(#path-2-inside-1_630_4298)"
          stroke="black"
          stroke-width="2.8"
        />
        <path
          d="M16 17C16 17.5253 15.8965 18.0454 15.6955 18.5307C15.4945 19.016 15.1999 19.457 14.8284 19.8284C14.457 20.1999 14.016 20.4945 13.5307 20.6955C13.0454 20.8965 12.5253 21 12 21C11.4747 21 10.9546 20.8965 10.4693 20.6955C9.98396 20.4945 9.54301 20.1999 9.17157 19.8284C8.80014 19.457 8.5055 19.016 8.30448 18.5307C8.10346 18.0454 8 17.5253 8 17L12 17H16Z"
          fill="black"
        />
        <path
          d="M7 18H17V19H7V18Z"
          fill="white"
        />
      </svg>
      <p
        class="m-0 flex-1"
      >
        Notifications
      </p>
      <div
        class="badge badge--red badge"
        role="badge"
      >
        +9
      </div>
    </a>
    <a
      class="root link"
      href="/web/articles"
    >
      <svg
        fill="none"
        height="20"
        role="icon"
        viewBox="0 0 24 24"
        width="20"
      >
        <path
          d="M12 4H5.77778C5.30628 4 4.8541 4.1873 4.5207 4.5207C4.1873 4.8541 4 5.30628 4 5.77778V18.2222C4 18.6937 4.1873 19.1459 4.5207 19.4793C4.8541 19.8127 5.30628 20 5.77778 20H18.2222C18.6937 20 19.1459 19.8127 19.4793 19.4793C19.8127 19.1459 20 18.6937 20 18.2222V12"
          stroke="black"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.6"
        />
        <path
          d="M17.9227 3.52798C18.2607 3.18992 18.7193 3 19.1973 3C19.6754 3 20.134 3.18992 20.472 3.52798C20.8101 3.86605 21 4.32456 21 4.80265C21 5.28075 20.8101 5.73926 20.472 6.07732L12.3991 14.1502L9 15L9.84978 11.6009L17.9227 3.52798Z"
          stroke="black"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="1.6"
        />
      </svg>
      <p
        class="m-0 flex-1"
      >
        Articles
      </p>
    </a>
  </div>
</body>
`;
